tools/run_tests/xds_k8s_test_driver/framework/xds_k8s_testcase.py

```diff
@@ -286,12 +288,22 @@ def assertRpcStatusCodes(self,
         diff_stats = self.diffAccumulatedStatsPerMethod(before_stats,
                                                         after_stats)
         stats = diff_stats.stats_per_method[method]
-        status = status_code.value[0]
-        for found_status, count in stats.result.items():
-            if found_status != status and count > stray_rpc_limit:
-                self.fail(f"Expected only status {status} but found status "
-                          f"{found_status} for method {method}:\n{diff_stats}")
-        self.assertGreater(stats.result[status_code.value[0]], 0)
+        for found_status_int, count in stats.result.items():
+            found_status = helpers_grpc.status_from_int(found_status_int)
+            if found_status != expected_status and count > stray_rpc_limit:
```

Comment:
[Nit] `found_status is not expected_status`?

Reply:
Yea, I was going for this, but I'm a bit scared of doing `is` with enums after an issue we had with the same enum `is` check failed, and `==` passed. Don't remember which one or if this was fixed. @gnossen may remember more.

Reply:
I'd say let's leave it for now to not re-run the tests for this change. It works as is. I can update it later.